src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs

```diff
@@ -84,6 +82,62 @@ public int TryGetLast(out bool found)
                 found = true;
                 return _end - 1;
             }
+
+            // Destination *must* be non-empty and exactly match the range length
+            private void InitializeSpan(Span<int> destination)
```

Comment:
This won't be inlined.  I'd expect this to regress for very short ranges due to the extra method call that wasn't there before.